benchmarks/src/main/scala/zio/internal/MailboxPollBenchmark.scala

```diff
@@ -0,0 +1,43 @@
+package zio.internal
+
+import org.openjdk.jmh.annotations._
+
+import java.util.concurrent.TimeUnit
+
+/**
+ * Defines a benchmark to measure the performance of dequeue operation of MPSC
+ * queues in the absence of concurrent operations.
+ */
+@BenchmarkMode(Array(Mode.AverageTime))
+@Fork(1)
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+@State(Scope.Thread)
+@Measurement(iterations = 5, time = 100, timeUnit = TimeUnit.MILLISECONDS)
+@Warmup(iterations = 5, time = 100, timeUnit = TimeUnit.MILLISECONDS)
+class MailboxPollBenchmark {
+
+  @Param(Array("0", "1000000"))
+  var fill: Int = _
+
+  @Param(
+    Array(
+      "ConcurrentLinkedQueue",
+      "Mailbox",
+      "MpscLinkedQueue"
+    )
+  )
+  var queue: String = _
+
+  var q: MailboxWrapper[AnyRef] = _
+
+  @Setup(Level.Iteration)
+  def setup() = {
+    q = MailboxWrapper(queue)
+    val element = new Object()
+    for (_ <- 0 to fill) q.add(element)
+  }
+
+  @Benchmark
+  def poll() =
+    q.poll()
```

Comment:
does this provide enough 'meat' for jmh? seems like stretching the definition of **micro** benchmark, what's the error jmh reports for this?

consider replacing with/adding a benchmark performing few dozens of polls followed by an isEmpty, I suspect this is closer to the actual usecase

Reply:
[poll.json](https://github.com/user-attachments/files/15845166/poll.json)
See attached for error values.

I will add an `OperationsPerInvocation` annotation.

Reply:
I have removed this benchmark since it does not add any real value.